src/fava/core/ingest.py

```diff
@@ -443,16 +447,37 @@ def extract(self, filename: str, importer_name: str) -> list[Directive]:
 
         try:
             path = Path(filename)
+            importer = self.importers[importer_name]
             new_entries = extract_from_file(
-                self.importers[importer_name],
+                importer,
                 path,
                 existing_entries=self.ledger.all_entries,
             )
         except Exception as exc:
             raise ImporterExtractError from exc
 
-        new_entries_list = [(filename, new_entries)]
         for hook_fn in self.hooks:
+            new_entries_list: HookOutput = []
+            try:
+                annotations = get_annotations(hook_fn)
+            except NameError:
+                annotations = {}
+            if (
+                "imported_entries" in annotations
+                and "Importer" in annotations["imported_entries"]
+            ):
+                importer_info = importer.file_import_info(path)
+                new_entries_list = [
+                    (
+                        filename,
+                        new_entries,
+                        importer_info.account,
+                        importer.importer,
+                    )
+                ]
+            else:
+                new_entries_list = [(filename, new_entries)]
+
             new_entries_list = hook_fn(
```

Comment:
this logic currently does not work correctly for multiple hooks - for each hook_fn, we'll have to have this check

Reply:
You're absolutely right, I adjusted the logic further